### ISSUE TITLE
Mismatch between arguments for auth0_client_grant 

### ISSUE THREAD

**Issue by peter-svensson** [auth0/terraform-provider-auth0#884]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

When creating a resource `auth0_client_grant` there seems to be a mismatch in the required configuration parameters and
the parameters actually passed to the Auth0 API.
The following config will result in errors when calling the API:

```hcl
resource "auth0_client_grant" "spa" {
  client_id = auth0_client.spa.id
  audience  = auth0_resource_server.backend.identifier
  scopes    = []
}
```

```shell
module.auth0.auth0_client_grant.spa: Creating...
│ Error: 400 Bad Request: Payload validation error: 'Missing required property: scope'.
│
│   with module.auth0.auth0_client_grant.spa,
│   on ../modules/auth0/apis.tf line 10, in resource "auth0_client_grant" "spa":
│   10: resource "auth0_client_grant" "spa" {
│
```

And `scope` is not a valid configuration parameter for `auth0_client_grant`:

```shell
│ Error: Unsupported argument
│
│   on ../modules/auth0/apis.tf line 14, in resource "auth0_client_grant" "spa":
│   14:   scope    = []
│
│ An argument named "scope" is not expected here.
╵
```

### Expectation

Either use `scope` or `scopes` (where `scopes` is probably the correct term) and make sure that the API call uses the correct argument (`scope` as described [here](https://auth0.com/docs/api/management/v2/client-grants/post-client-grants)

### Reproduction

See description.

### Auth0 Terraform Provider version

1.1.0

### Terraform version

1.6.4

**Comment by ansraliant** [comment]
Can confirm it's happening on `1.1.0`

Quick work around is to go back to `1.0.0` until this is fixed

**Comment by sergiught** [comment]
Hello everyone 👋🏻,

Thank you for bringing this to our attention. The issue doesn't stem from a mismatch between the terraform config scopes and the API scope attribute. Occasionally, we enhance the developer experience by refining the naming conventions used in the API. Behind the scenes, the resource correctly maps and makes API calls.

The current issue arises because the Go SDK we utilize doesn't allow the transmission of scopes as an empty array. However, this issue has been addressed in https://github.com/auth0/go-auth0/pull/318. As soon as this fix is released, we will incorporate it into the provider. In the meantime, kindly ensure that client grants with non-empty scopes are created.

Thank you for your understanding and patience.

**Comment by willvedd** [comment]
Fixed with [v1.1.1](https://github.com/auth0/terraform-provider-auth0/releases/tag/v1.1.1). Thank you for submitting this issue!
